x-pack/plugins/files/public/components/upload_file/upload_file.test.tsx

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { act } from 'react-dom/test-utils';
+import { registerTestBed } from '@kbn/test-jest-helpers';
+
+import {
+  FileKindsRegistryImpl,
+  setFileKindsRegistry,
+  getFileKindsRegistry,
+} from '../../../common/file_kinds_registry';
+
+import { createMockFilesClient } from '../../mocks';
+
+import { FileJSON } from '../../../common';
+import { FilesContext } from '../context';
+import { UploadFile, Props } from './upload_file';
+import { UploadFileUI } from './components';
+
+describe('UploadFile', () => {
+  const sleep = (ms: number) => new Promise((res) => setTimeout(res, ms));
+  let onDone: jest.Mock;
+  let onError: jest.Mock;
+  let client: ReturnType<typeof createMockFilesClient>;
+
+  async function initTestBed(props?: Partial<Props>) {
+    const createTestBed = registerTestBed((p: Props) => (
+      <FilesContext>
+        <UploadFile {...p} />
+      </FilesContext>
+    ));
+
+    const testBed = await createTestBed({
+      client,
+      kind: 'test',
+      onDone,
+      onError,
+      ...props,
+    });
+
+    const baseTestSubj = `filesUploadFile`;
+
+    const testSubjects = {
+      base: baseTestSubj,
+      uploadButton: `${baseTestSubj}.uploadButton`,
+      retryButton: `${baseTestSubj}.retryButton`,
+      cancelButton: `${baseTestSubj}.cancelButton`,
+      errorMessage: `${baseTestSubj}.error`,
+    };
+
+    return {
+      ...testBed,
+      actions: {
+        addFiles: (files: File[]) =>
+          act(async () => {
+            testBed.component.find(UploadFileUI).props().onChange(files);
+            await sleep(1);
+            testBed.component.update();
+          }),
+        upload: (retry = false) =>
+          act(async () => {
+            testBed
+              .find(retry ? testSubjects.retryButton : testSubjects.uploadButton)
+              .simulate('click');
+            await sleep(1);
+            testBed.component.update();
+          }),
+        abort: () =>
+          act(() => {
+            testBed.find(testSubjects.cancelButton).simulate('click');
+            testBed.component.update();
+          }),
+        wait: (ms: number) =>
+          act(async () => {
+            await sleep(ms);
+            testBed.component.update();
+          }),
+      },
+      testSubjects,
+    };
+  }
+
+  beforeAll(() => {
```

Comment:
Should we add more test cases here? A successful upload scenario, retry scenario etc.